src/index.js

```diff
@@ -0,0 +1,284 @@
+import {BigNumber} from 'bignumber.js';
+
+// http://stackoverflow.com/questions/3885817/how-to-check-if-a-number-is-float-or-integer
+const isInt = (maybeInt) => maybeInt % 1 === 0;
+
+const compareCentsFunction = (comparator, Class) =>
+  function (val, options) {
+    if (options == null) {
+      options = {};
+    }
+    const cents = this;
+    if (options.strict == null) {
+      options.strict = true;
+    }
+    if (options.strict) {
+      return val instanceof Class && comparator(cents, val);
+    }
+    const otherCents = new Class(val);
+    return comparator(cents, otherCents);
+  };
+
+const comparators = {
+  equals(cents, otherCents) {
+    return cents.toNumber() === otherCents.toNumber();
+  },
+  lessThan(cents, otherCents) {
+    return cents.toNumber() < otherCents.toNumber();
+  },
+  lessThanOrEqual(cents, otherCents) {
+    return cents.toNumber() <= otherCents.toNumber();
+  },
+  greaterThan(cents, otherCents) {
+    return cents.toNumber() > otherCents.toNumber();
+  },
+  greaterThanOrEqual(cents, otherCents) {
+    return cents.toNumber() >= otherCents.toNumber();
+  },
+};
+
+// This method supports static method calls of the form:
+//   Cents.staticMethod(...)
+//
+// The ... must include at least one argument.
+//
+// Case 1: ... => single argument array
+//   There must be no other arguments and each value in the array must "pass" the
+//   validator (boolean) predicate function. Returns the array.
+// Case 2: ... => multiple argument values
+//   Each value must "pass" the validator predicate function. Returns an array of the values.
+const arrayifySplat = function (splat, validator) {
+  if (!(splat.length > 0)) {
+    throw new Error('Expect at least one argument');
+  }
+
+  if (Array.isArray(splat[0])) {
+    if (splat.length !== 1) {
+      throw new Error('Expect a single array argument');
+    }
+    splat = splat[0];
+  }
+
+  if (validator != null) {
+    splat.forEach(function (val) {
+      if (!validator(val)) {
+        throw new Error(`Unexpected value ${val}`);
+      }
+    });
+  }
+
+  return splat;
+};
+class Cents {
+  constructor(value) {
+    this.equals = compareCentsFunction(comparators.equals, Cents);
+    this.lessThan = compareCentsFunction(comparators.lessThan, Cents);
```

Comment:
I just realized – an even simpler solution here...
```suggestion
    this.lessThan = compareCentsFunction(comparators.lessThan, Cents);
    this.lt = this.lessThan;
```
etc.